File: src/main/java/com/wangweimin/learnspark/utils/SparkSessionUtil.java
package com.wangweimin.learnspark.utils;

import org.apache.spark.sql.SparkSession;

/**
 * SparkSession工具类。
 * @author Phoenix Lee (liqingyuan1986@aliyun.com)
 * @since 2016/9/19
 */
public class SparkSessionUtil {

    public static SparkSession getSpark() {
        String warehouseLocation = "file:" + System.getProperty("user.dir") + "spark-warehouse";
        return SparkSession
                .builder()
                .appName("Java Spark Hive Example")
                .config("spark.sql.warehouse.dir", warehouseLocation)
                .enableHiveSupport()
                .getOrCreate();
    }
}

File: src/main/scala/utils/SparkSessionUtil.scala
package utils

import org.apache.spark.sql.SparkSession

object SparkSessionUtil {
  def getInstance(appName:String): SparkSession ={
    val sparkSession = SparkSession
      .builder()
      .appName(appName)
      .master("local[*]")
      .config("spark.some.config.option", "some-value")
      .getOrCreate()
    return sparkSession;
  }
}
